src/track/cue.h

```diff
@@ -89,4 +107,57 @@ class CuePointer: public std::shared_ptr<Cue> {
     }
 };
 
+class CuePosition {
+  public:
+    CuePosition()
+        : m_position(0.0), m_source(Cue::UNKNOWN) {}
+    CuePosition(double position, Cue::CueSource source)
+        : m_position(position), m_source(source) {}
+
+    double getPosition() const {
+        return m_position;
+    }
+
+    void setPosition(double position) {
+        m_position = position;
+    }
+
+    Cue::CueSource getSource() const {
+        if (m_position == 0.0 || m_position == -1.0) {
```

Comment:
A getter should not contain any logic and just return part of the current state, i.e. m_source.